src/middlewared/middlewared/plugins/vm.py

```diff
@@ -1987,6 +2154,57 @@ def translate_device(dev):
         else:
             return False
 
+    @private
+    def get_iommu_type(self):
+        async def check_util_output(proc, pattern):
+            found = False
+            while True:
+                line = await proc.stdout.readline()
+                if line:
+                    if re.search(pattern, line.decode('utf-8')):
+                        found = True
+                else:
+                    break  # Reached EOF
+            return found
+
+        async def ext_util(iommu_type, cmd_args, pattern):
+            found = False
+            proc = None
+            try:
+                proc = await asyncio.create_subprocess_exec(*cmd_args,
+                                                            stdout=asyncio.subprocess.PIPE,
+                                                            stderr=asyncio.subprocess.DEVNULL)
+                found = await asyncio.wait_for(check_util_output(proc, pattern), timeout=5)
+            except (asyncio.TimeoutError, OSError, ValueError) as e:
+                if proc is not None:
+                    proc.kill()
+                self.middleware.logger.error('An error occured when checking for iommu '
+                                             f'({iommu_type}): {e}.')
+            else:
+                if proc.returncode == 0:
+                    return found
+                else:
+                    self.middleware.logger.error('An error occured when checking for iommu '
+                                                 f'({iommu_type}): Subprocess ({cmd_args}) '
+                                                 f'exited with status {proc.returncode}.')
+            return None
+
+        async def int_sysctl(iommu_type, iou):
+            res = sysctl.filter(iou)
```

Comment:
`sysctl` module should be used in synchronous functions, I am sorry I should have clarified in my last comment